plugins/action/system/shutdown.py

```diff
@@ -0,0 +1,234 @@
+
+    def perform_shutdown(self, task_vars, distribution):
+        result = {}
+        shutdown_result = {}
+        shutdown_command = self.get_shutdown_command(task_vars, distribution)
+        shutdown_command_args = self.get_shutdown_command_args(distribution)
+        shutdown_command_exec = '{0} {1}'.format(shutdown_command, shutdown_command_args)
+
+        self.cleanup(force=True)
+        try:
+            display.vvv("{action}: shutting down server...".format(action=self._task.action))
+            display.debug("{action}: shutting down server with command '{command}'".format(action=self._task.action, command=shutdown_command_exec))
+            shutdown_result = self._low_level_execute_command(shutdown_command_exec, sudoable=self.DEFAULT_SUDOABLE)
+        except AnsibleConnectionFailure as e:
+            # If the connection is closed too quickly due to the system being shutdown, carry on
+            display.debug('{action}: AnsibleConnectionFailure caught and handled: {error}'.format(action=self._task.action, error=to_text(e)))
+            shutdown_result['rc'] = 0
+
+        if shutdown_result['rc'] != 0:
+            result['failed'] = True
+            result['shutdown'] = False
+            result['msg'] = "Shutdown command failed. Error was {stdout}, {stderr}".format(
+                stdout=to_native(shutdown_result['stdout'].strip()),
+                stderr=to_native(shutdown_result['stderr'].strip()))
+            return result
+
+        result['failed'] = False
+        return result
+
+    def run(self, tmp=None, task_vars=None):
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        # If running with local connection, fail so we don't shutdown ourself
+        if self._connection.transport == 'local':
+            msg = 'Running {0} with local connection would shutdown the control node.'.format(self._task.action)
+            return {'changed': False, 'elapsed': 0, 'shutdown': False, 'failed': True, 'msg': msg}
+
+        if self._play_context.check_mode:
+            return {'changed': True, 'elapsed': 0, 'shutdown': True}
+
+        if task_vars is None:
+            task_vars = {}
+
+        self.deprecated_args()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        if result.get('skipped', False) or result.get('failed', False):
+            return result
+
+        distribution = self.get_distribution(task_vars)
+
+        # Get the original connection_timeout option var so it can be reset after
+        original_connection_timeout = None
+        try:
+            original_connection_timeout = self._connection.get_option('connection_timeout')
+            display.debug("{action}: saving original connect_timeout of {timeout}".format(action=self._task.action, timeout=original_connection_timeout))
+        except KeyError:
+            display.debug("{action}: connect_timeout connection option has not been set".format(action=self._task.action))
```

Comment:
```suggestion
```
I don't think this is needed anymore.

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/700/commits/83b05b476ab4930c1054ed7bb4126bf818a4e0f3.